cmd/agent/entrypoint.go

```diff
@@ -352,6 +352,22 @@ func (ep *Entrypoint) TriggerReload() bool {
 	return true
 }
 
+// Triggers a reload of the config on each tick of the ticker until the context
+// completes.
+func (ep *Entrypoint) PollConfig(ctx context.Context, t *time.Ticker) error {
```

Comment:
This doesn't need to be an exposed function :) I'd like for us to try to keep the public API as small as we can.